### ISSUE TITLE
forEach on a RegExp match array randomly skips last element

### ISSUE THREAD

**Issue by seishun** [nodejs/node-v0.x-archive#8970]
```js
function doStuff() {
  var regexPattern = /(a)|(a)/;
  
  var match = regexPattern.exec('a');
  var i = 0;
  match.forEach(function(group) {
    i++;
  });
  if (i != match.length) {
    console.log('i ' + i + ', len ' + match.length);
    throw new Error('wtf?');
  }
}

for (var i = 0; i < 100000; i++) { doStuff(); }
```

I have been unable to simplify it further. Tested with v0.10.35 on Linux with a self-built binary and on Windows using the provided binary.

**Comment by micnic** [comment]
Very interesting issue, I also discovered it happens on the 1827th iteration of the loop, this seems to be a V8 problem, which is fixed in node's 0.11 releases.

**Comment by brendanashworth** [comment]
According to https://github.com/nodejs/node-v0.x-archive/issues/8970#issuecomment-68708060, it is a v8 issue that is fixed in recent releases (0.10 wouldn't receive a v8 backport), so I'll close this for now.
